crates/compiler/src/revive_resolc.rs

```diff
@@ -1,2 +1,49 @@
-//! Implements the [crate::SolidityCompiler] trait with resolc for
-//! compiling contracts to PVM bytecode.
+//! Implements the [SolidityCompiler] trait with `resolc` for
+//! compiling contracts to PolkaVM (PVM) bytecode.
+
+use std::{
+    path::PathBuf,
+    process::{Command, Stdio},
+};
+
+use crate::{CompilerInput, CompilerOutput, SolidityCompiler};
+use revive_solc_json_interface::SolcStandardJsonOutput;
+
+/// A wrapper around the `resolc` binary, emitting PVM-compatible bytecode.
+pub struct Resolv {
+    /// Path to the `resolc` executable
+    resolc_path: PathBuf,
+}
+
+impl SolidityCompiler for Resolv {
+    /// No extra compiler options for now.
+    type Options = ();
```

Comment:
Resolc knows a bunch of specific command line options, for example  the solc path to use which we will require. The command line arguments structs of resolc would be nice. However might be overkill. For now it would just do to have a `Vec<String>` as the type and then those can be added to the command as args directly.